src/core/App.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * @author Lewy Blue / https://github.com/looeee
+ *
+ * parameters = {
+ *
+ *  canvas: <canvas> DOM element,
+ *
+ *  scene: new THREE.Scene(),
+ *
+ *  camera: new THREE.Camera( ... )
+ *
+ *  renderer: new THREE.WebGLRenderer( ... )
+ *
+ *  autoResize: <bool>
+ *
+ * }
+ */
+
+function App( parameters ) {
+
+	parameters = parameters || {};
+
+	if ( parameters.canvas !== undefined ) {
+
+		this.canvas = parameters.canvas;
+
+	} else {
+
+		this.canvas = document.body.appendChild( document.createElement( 'canvas' ) );
+		this.canvas.style.width = this.canvas.style.height = '100%';
```

Comment:
Are you sure? The idea is that the default canvas gets set to 100% of the body size. If the body is set to 50% of the window width then canvas will also be 50% of window width. Just tested it and seems to be working as expected.